src/daemon/jsonrpc/api.rs

```diff
@@ -1021,13 +1025,61 @@ impl RpcApi for RpcImpl {
         Ok(json!({}))
     }
 
-    fn listspendtxs(&self, meta: Self::Metadata) -> jsonrpc_core::Result<serde_json::Value> {
+    fn listspendtxs(
+        &self,
+        meta: Self::Metadata,
+        status: Option<ListSpendStatus>,
+    ) -> jsonrpc_core::Result<serde_json::Value> {
         manager_only!(meta);
 
-        let db_path = meta.rpc_utils.revaultd.read().unwrap().db_file();
+        let revaultd = meta.rpc_utils.revaultd.read().unwrap();
+        let db_path = revaultd.db_file();
+
         let spend_tx_map = db_list_spends(&db_path).map_err(|e| internal_error!(e))?;
         let mut listspend_entries = Vec::with_capacity(spend_tx_map.len());
         for (_, (psbt, deposit_outpoints)) in spend_tx_map {
+            // Filter by status
+            if let Some(s) = &status {
+                // Let's retrieve the vault statuses for the outpoints spent
+                let vault_statuses: Vec<_> = deposit_outpoints
+                    .iter()
+                    .map(|o| db_vault_by_deposit(&db_path, &o))
+                    .collect::<Result<Vec<_>, _>>()
+                    .map_err(|e| internal_error!(e))?
+                    .into_iter()
+                    .map(|v| v.expect("It's the deposit outpoint of a spend").status)
+                    .collect();
+                let status = if vault_statuses.iter().all(|v| *v == VaultStatus::Spent) {
+                    // Well, if all of them are in a `Spent` status then the spend is Confirmed
+                    // FIXME: Should I also check that the transaction is in blockchain with getwallettransaction,
+                    // in case another Spend transaction spent this vaults? How do I do that, as my
+                    // spend_tx might not be final? (I might not have all the signatures)
+                    ListSpendStatus::Confirmed
+                } else if vault_statuses.iter().any(|v| *v == VaultStatus::Canceled) {
+                    // Otherwise, are they canceled? (One canceled is enough for canceling the
+                    // whole spend)
+                    ListSpendStatus::Canceled
+                } else if vault_statuses
+                    .iter()
+                    .all(|v| *v == VaultStatus::Unvaulting || *v == VaultStatus::Unvaulted)
+                {
+                    // Are we awaiting for the unvault CSV?
+                    ListSpendStatus::Ongoing
+                } else if vault_statuses.iter().all(|v| *v == VaultStatus::Spending) {
+                    // At least, did we broadcast it?
+                    ListSpendStatus::Broadcasted
+                } else if let Ok(()) = psbt.clone().finalize(&revaultd.secp_ctx) {
+                    ListSpendStatus::Final
+                } else {
+                    // FIXME: Maybe here the tx is final, but another manager has all the
+                    // signatures, and our local version is still non final
+                    ListSpendStatus::NonFinal
+                };
```

Comment:
Hmm.. It seems quite convoluted to me to retrieve the statuses of the vaults a Spend is unvaulting.